connectors/db/vector.py

```diff
@@ -293,13 +498,16 @@ def search(self, query, agent_id: int):
         for provider in self.search_providers:
             results.extend(provider.search(query, search_filter))
 
+        # Remove duplicates based on text similarity
+        deduped_results = self.deduplicate_results(results)
+
         # Sort the results by relevance score
-        results.sort(key=lambda result: result.score, reverse=True)
+        sorted_results = self.rank_results_with_llm(query, deduped_results)
```

Comment:
Can we feature flag this with a config env var so that we can turn it off if there's a problem or significant lag

Reply:
Done!